providers/fips/fipsprov.c

```diff
@@ -37,19 +38,13 @@ static OSSL_FUNC_provider_query_operation_fn fips_query;
 #define ALG(NAMES, FUNC) ALGC(NAMES, FUNC, NULL)
 
 extern OSSL_FUNC_core_thread_start_fn *c_thread_start;
-int FIPS_security_check_enabled(void);
+int FIPS_security_check_enabled(OSSL_LIB_CTX *libctx);
 
 /*
- * TODO(3.0): Should these be stored in the provider side provctx? Could they
- * ever be different from one init to the next? Unfortunately we can't do this
- * at the moment because c_put_error/c_add_error_vdata do not provide
- * us with the OSSL_LIB_CTX as a parameter.
+ * Should these function pointers be stored in the provider side provctx? Could
```

Comment:
This is the reason I didn't try doing this PR myself :) Passing the libctx to these would be painful.

Reply:
When setting the function pointers, should we check the values and refuse loading the provider if the current value is non-NULL and different from the new value?

Reply:
I've added a sanity check to ensure the up-call function pointers haven't changed.

Reply:
I think that's good enough for 3.0.